### PR TITLE
[Pipeline] animatediff + vid2vid + controlnet

### PR DESCRIPTION
**Pull request by reallyigor**
# What does this PR do?
This PR adds ControlNet support to the Video To Video AnimateDiff.

# Fixes:
See #9326 `[Pipeline] AnimateDiff + VideoToVideo + ControlNet #9326`

# Results:
## Default pipeline:
|          ControlNet          | Strength 0.1                                                                                     | Strength 0.5                                                                                     | Strength 0.8                                                                                     |
|---------------------|--------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------|
|          Depth               | Strength 0.1 | Strength 0.5 | Strength 0.8 |
|          OpenPose            | Strength 0.1_2 | Strength 0.5_2 | Strength 0.8_2 |

## An example with IPAdapter usage alongside ControlNet

| Input | IP Adapter  | Result | Without IP Adapter |
|-------|-----------------|----------------------------|----|
|austranaut|ip-adapter-plus-_sd15| animatediff_vid2vid_controlnet_0 8_ipa_2 | animatediff_vid2vid_controlnet_0 8_2 |
|woman|ip-adapter-plus-face_sd15 | animatediff_vid2vid_controlnet_0 8_ipa | animatediff_vid2vid_controlnet_0 8_2 |

## Prompt travel on a tik-tok dance video
```python
strength = 0.8
pipe.set_ip_adapter_scale(0.0)

context_length = 16
context_stride = 4
pipe.enable_free_noise(context_length=context_length, context_stride=context_stride)

# Can be a single prompt, or a dictionary with frame timesteps
prompt = {
    0: "a girl on a winter day, sparkly leaves in the background, snow flakes, close up",
    10: "a girl on a autumn day, yellow leaves in the background, close up",
    20: "a girl on a rainy day, tropical leaves in the background, close up",
}
negative_prompt = "bad quality, worst quality"

with torch.inference_mode():
    video = pipe(
        video=video,
        prompt=prompt,
        negative_prompt=negative_prompt,
        num_inference_steps=10,
        guidance_scale=2.0,
        controlnet_conditioning_scale=0.75,
        conditioning_frames=conditioning_frames,
        strength=strength,
        generator=torch.Generator().manual_seed(42),
        ip_adapter_image=ip_adapter_image,
    ).frames[0]
 ```
prompt_travel_0 8_2

# How to test:
```python
import torch
from PIL import Image
from tqdm.auto import tqdm

from controlnet_aux.processor import OpenposeDetector
from diffusers import AnimateDiffVideoToVideoControlNetPipeline
from diffusers.utils import export_to_gif, load_video
from diffusers import AutoencoderKL, ControlNetModel, MotionAdapter, LCMScheduler

controlnet = ControlNetModel.from_pretrained("lllyasviel/sd-controlnet-openpose", torch_dtype=torch.float16)
motion_adapter = MotionAdapter.from_pretrained("wangfuyun/AnimateLCM")
vae = AutoencoderKL.from_pretrained("stabilityai/sd-vae-ft-mse", torch_dtype=torch.float16)

pipe = AnimateDiffVideoToVideoControlNetPipeline.from_pretrained(
    "SG161222/Realistic_Vision_V5.1_noVAE",
    motion_adapter=motion_adapter,
    controlnet=controlnet,
    vae=vae,
).to(device="cuda", dtype=torch.float16)

pipe.scheduler = LCMScheduler.from_config(pipe.scheduler.config, beta_schedule="linear")
pipe.load_lora_weights("wangfuyun/AnimateLCM", weight_name="AnimateLCM_sd15_t2v_lora.safetensors", adapter_name="lcm-lora")
pipe.set_adapters(["lcm-lora"], [0.8])

video = load_video("dance.gif")
video = [frame.convert("RGB") for frame in video]

prompt = "astronaut in space, dancing"
negative_prompt = "bad quality, worst quality, jpeg artifacts, ugly"

open_pose = OpenposeDetector.from_pretrained("lllyasviel/Annotators").to("cuda")

conditioning_frames = []
for frame in tqdm(video):
    conditioning_frames.append(open_pose(frame))

strength = 0.8
with torch.inference_mode():
    video = pipe(
        video=video,
        prompt=prompt,
        negative_prompt=negative_prompt,
        num_inference_steps=10,
        guidance_scale=2.0,
        controlnet_conditioning_scale=0.75,
        conditioning_frames=conditioning_frames,
        strength=strength,
        generator=torch.Generator().manual_seed(42),
    ).frames[0]

video = [frame.resize(conditioning_frames[0].size) for frame in video]
export_to_gif(video, f"animatediff_vid2vid_controlnet.gif", fps=8)
```

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

I created new tests by adapting ones from the regular AnimateDiff Video-to-Video pipeline.
```python
pytest tests/pipelines/animatediff/test_animatediff_video2video_controlnet.py
```

My code fails only on one test:
```python
test_from_pipe_consistent_forward_pass_cpu_offload
```
```bash
========================================= 1 failed, 37 passed, 2 skipped, 37 warnings in 42.66s ==========================================
```

The problem is that the original (that is already in Diffusers) AnimateDiff Video To Video also fails on this test. I couldn't identify the problem. These are the results of the original AnimateDiff Video To Video:
```python
pytest tests/pipelines/animatediff/test_animatediff_video2video.py 
```
```bash
========================================= 1 failed, 37 passed, 2 skipped, 37 warnings in 40.86s ==========================================
```

## Who can review?

@DN6 @a-r-r-o-w

### CODE REVIEW

File: src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py
Comment (a-r-r-o-w) [inline src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py:271]: Could a `# Copied from` comment be applicable here? If so, which should be the case if it matches 1:1 with another pipeline, could you add one?

File: src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py
Comment (a-r-r-o-w) [inline src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py:524]: Same comment about `# Copied from` here

File: src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py
Comment (a-r-r-o-w) [inline src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py:739]: Same comment about `# Copied from` here

File: src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py
Comment (a-r-r-o-w) [inline src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py:748]: Same comment about `# Copied from` here

File: src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py
Comment (a-r-r-o-w) [inline src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py:1247]:
```suggestion
            # 10. Denoising loop
```

File: docs/source/en/api/pipelines/animatediff.md
Comment (a-r-r-o-w) [inline docs/source/en/api/pipelines/animatediff.md:32]: Maybe an example with this pipeline could also be added here?

File: src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py
Comment (a-r-r-o-w) [inline src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py:77]: Could we replace this with the following URL: https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/animatediff-vid2vid-input-1.gif

and update the example to using the panda prompt? Alternatively, if you'd like to use `dance.gif` here, could you open a PR to https://huggingface.co/datasets/huggingface/documentation-images/

**Review by a-r-r-o-w — COMMENTED** [review]
Thank you very much, this is looking really good! I would like to and appreciate seeing a few experimental results though, just to verify functionality:
- An example making use of `enforce_inference_steps=True`, possibly latent upscale from the other reference PRs or anything unique of your choice
- An example making use of the FreeNoise prompt travel feature
- An example with IPAdapter usage alongside ControlNet

By looking at the diffs, all changes look great to me :100:

File: src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py
Comment (RedHash) [reply src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py:77]: I think an example with panda is a bit boring, so I opened a PR with the dancing girl .gif 
https://huggingface.co/datasets/huggingface/documentation-images/discussions/360
Could you, please, merge it? @a-r-r-o-w

File: docs/source/en/api/pipelines/animatediff.md
Comment (RedHash) [reply docs/source/en/api/pipelines/animatediff.md:32]: Yes, I just didnt know how to add my own .gif to diffusers. Now I opened a PR to the huggingface documentation

**Comment by RedHash** [comment]
> * An example making use of `enforce_inference_steps=True`, possibly latent upscale from the other reference PRs or anything unique of your choice

Can you please give me a hint about where I can find an example code?

> * An example making use of the FreeNoise prompt travel feature

Same^

File: src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py
Comment (RedHash) [reply src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py:271]: Added ✅

File: src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py
Comment (RedHash) [reply src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py:524]: Added ✅

File: src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py
Comment (RedHash) [reply src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py:739]: Added ✅

File: src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py
Comment (RedHash) [reply src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py:748]: Added ✅

File: src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py
Comment (RedHash) [reply src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py:1247]: Fixed ✅

**Comment by RedHash** [comment]
IP-Adapter test

| Input | IP Adapter  | Result | Without IP Adapter |
|-------|-----------------|----------------------------|----|
|austranaut|ip-adapter-plus-_sd15| animatediff_vid2vid_controlnet_0 8_ipa_2 | animatediff_vid2vid_controlnet_0 8_2 |
|woman|ip-adapter-plus-face_sd15 | animatediff_vid2vid_controlnet_0 8_ipa | animatediff_vid2vid_controlnet_0 8_2 |

**Comment by RedHash** [comment]
Prompt travel
```python
strength = 0.8
pipe.set_ip_adapter_scale(0.0)

context_length = 16
context_stride = 4
pipe.enable_free_noise(context_length=context_length, context_stride=context_stride)

# Can be a single prompt, or a dictionary with frame timesteps
prompt = {
    0: "an austonaut on a winter day, sparkly leaves in the background, snow flakes, close up",
    6: "an austonaut on a autumn day, yellow leaves in the background, close up",
    12: "an austonaut on a rainy day, tropical leaves in the background, close up",
}
negative_prompt = "bad quality, worst quality"

with torch.inference_mode():
    video = pipe(
        video=video,
        prompt=prompt,
        negative_prompt=negative_prompt,
        num_inference_steps=10,
        guidance_scale=2.0,
        controlnet_conditioning_scale=0.75,
        conditioning_frames=conditioning_frames,
        strength=strength,
        generator=torch.Generator().manual_seed(42),
        ip_adapter_image=ip_adapter_image,
    ).frames[0]
```
prompt_travel

**Comment by RedHash** [comment]
Latent upscaling also works, although, with the current hyperparams, the results are not great:

```python
strength = 0.8
pipe.set_ip_adapter_scale(0.0)

context_length = 16
context_stride = 4
pipe.enable_free_noise(context_length=context_length, context_stride=context_stride)

# Can be a single prompt, or a dictionary with frame timesteps
prompt = {
    0: "a girl on a winter day, sparkly leaves in the background, snow flakes, close up",
    10: "a girl on a autumn day, yellow leaves in the background, close up",
    20: "a girl on a rainy day, tropical leaves in the background, close up",
}
negative_prompt = "bad quality, worst quality"

with torch.inference_mode():
    latents = pipe(
        video=video,
        prompt=prompt,
        negative_prompt=negative_prompt,
        num_inference_steps=10,
        guidance_scale=2.0,
        controlnet_conditioning_scale=0.75,
        conditioning_frames=conditioning_frames,
        strength=strength,
        generator=torch.Generator().manual_seed(42),
        ip_adapter_image=ip_adapter_image,
        output_type="latent",
    ).frames

import torch.nn.functional as F

# Run latent upscaling
# Note that only naive upscaling is done here. Alternatively, a latent upscaler
# model could be used

batch_size, num_channels, num_frames, latent_height, latent_width = latents.shape
height = 512
width = 512
scale_factor = 1
scale_method = "nearest-exact"
upscaled_height = int(height * scale_factor)
upscaled_width = int(width * scale_factor)
upscaled_latent_height = int(latent_height * scale_factor)
upscaled_latent_width = int(latent_width * scale_factor)
strength = 0.5

upscaled_latents = []
for i in range(batch_size):
    latent = F.interpolate(latents[i], size=(upscaled_latent_height, upscaled_latent_width), mode="nearest-exact")
    upscaled_latents.append(latent.unsqueeze(0))
upscaled_latents = torch.cat(upscaled_latents, dim=0)

# Run pipeline for denoising upscaled latents
with torch.inference_mode():
    result = pipe(
        prompt=prompt,
        negative_prompt=negative_prompt,
        num_inference_steps=10,
        guidance_scale=2.0,
        controlnet_conditioning_scale=0.75,
        conditioning_frames=conditioning_frames,
        strength=strength,
        generator=torch.Generator().manual_seed(42),
        ip_adapter_image=ip_adapter_image,
        output_type="pil",
        latents=upscaled_latents,
    ).frames[0]
   
result = [frame.resize(conditioning_frames[0].size) for frame in result]
export_to_gif(result, "latent_upscaled.gif", fps=8)
```
latent_upscaled

File: src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py
Comment (RedHash) [reply src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py:77]: I reopened the PR properly, please, merge it
https://huggingface.co/datasets/huggingface/documentation-images/discussions/362

File: docs/source/en/api/pipelines/animatediff.md
Comment (RedHash) [reply docs/source/en/api/pipelines/animatediff.md:32]: I updated the example with the description and code. The only thing missing now is the result, which I'll add after this PR is merged: https://huggingface.co/datasets/huggingface/documentation-images/discussions/362

Comment (a-r-r-o-w) [reply docs/source/en/api/pipelines/animatediff.md:32]: thanks, merged

File: docs/source/en/api/pipelines/animatediff.md
Comment (a-r-r-o-w) [inline docs/source/en/api/pipelines/animatediff.md:558]: Could we use the download url from the HF documentation-images repo here? `load_video` can accept URLs too

File: docs/source/en/api/pipelines/animatediff.md
Comment (a-r-r-o-w) [inline docs/source/en/api/pipelines/animatediff.md:590]: Could you add it now?

File: src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py
Comment (a-r-r-o-w) [inline src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py:77]: Needs the HF docs URL here too

**Review by a-r-r-o-w — COMMENTED** [review]
LGTM. cc @DN6 if you're free to give this a look

File: src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py
Comment (RedHash) [reply src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video_controlnet.py]: Added  ✅

File: docs/source/en/api/pipelines/animatediff.md
Comment (RedHash) [reply docs/source/en/api/pipelines/animatediff.md:590]: Added ✅

File: docs/source/en/api/pipelines/animatediff.md
Comment (RedHash) [reply docs/source/en/api/pipelines/animatediff.md]: Added ✅

**Comment by RedHash** [comment]
I've updated the links to the input video and added an example output to the docs 🥳🥳

**Comment by a-r-r-o-w** [comment]
Could you run `make style` and `make fix-copies` so that the style tests pass? I would be careful with `make fix-copies` because if it has been used incorrectly by mistake somewhere, it will overwrite your changes.

**Comment by RedHash** [comment]
Sure, I used to run it before every commit, but I forgot last time. Done ✅

**Comment by a-r-r-o-w** [comment]
A couple of failing tests need to be addressed here before merge:

- We recently updated the ip-adapter tests in #9285. I think you just have to change the name of the test correctly
```
FAILED tests/pipelines/animatediff/test_animatediff_video2video_controlnet.py::AnimateDiffVideoToVideoControlNetPipelineFastTests::test_ip_adapter_single - AttributeError: 'super' object has no attribute 'test_ip_adapter_single'
```
- The failing test [here](https://github.com/huggingface/diffusers/actions/runs/10682579962/job/29608881043?pr=9337) is unrelated to the PR. If you update branch with `main`, I can restart the test run so this passes

**Comment by RedHash** [comment]
Thank you for clarification! I updated with main and changed the ip adapter test function 🙏  
@a-r-r-o-w

**Comment by RedHash** [comment]
It wasn't easy to understand what value to use for the `expected_pipe_slice` in the `test_ip_adapter`, but I figured it out :) 
@a-r-r-o-w

**Comment by a-r-r-o-w** [comment]
Hey, this is looking good. The failing test is due to the tests being run on different machine types. To get the correct numbers, we'd have to get them from the specific CPU runners we use. I can get them to you some time tomorrow or over the weekend since caught up with other things at the moment

**Comment by RedHash** [comment]
>To get the correct numbers, we'd have to get them from the specific CPU runners we use. I can get them to you some time tomorrow or over the weekend since caught up with other things at the moment

Thank you <3, I appreciate it!

**Comment by RedHash** [comment]
Offtop: Isn't it weird that CPU tests depend on the CPU type? 
This test passes on my machine :/

**Comment by RedHash** [comment]
@a-r-r-o-w 
please take a look when you have a moment 🙏

**Comment by a-r-r-o-w** [comment]
hey, sorry for the delay! here you go:

```
[0.5569  0.6250   0.4144 0.5613 0.5563  0.5213
 0.5091   0.4950 0.4950 0.5684 0.3858  0.4863
 0.6457  0.4311 0.5517 0.5608  0.4417  0.5377 ]
```

**Comment by reallyigor** [comment]
Thanks a lot, Aryan! 
I updated the test ✅

**Review by a-r-r-o-w — APPROVED** [review]
Very cool PR @reallyigor! Everything was correct for the most part from the very beginning, which is quite rare to come across so really great work :)

**Comment by reallyigor** [comment]
>Very cool PR @reallyigor! Everything was correct for the most part from the very beginning, which is quite rare to come across so really great work :)

Thank you for your kind words 😍 , I really appreciate it

**Comment by reallyigor** [comment]
What are the next steps for merging?

**Comment by a-r-r-o-w** [comment]
Nothing much, it looks great! I'm just waiting to check if @DN6 would like to give this a review too by Monday since it was his ask initially. Happy to merge by EOD tomorrow even if he isn't able to take a look because seems like results are as expected :)

**Comment by reallyigor** [comment]
🥳

**Comment by reallyigor** [comment]
@DN6 
Do you mind taking a look when you have a moment? 🙏

**Comment by HanLiii** [comment]
@a-r-r-o-w @reallyigor Thank you for the great work!
My diffusers lib is V0.30.3, I tried to import AnimateDiffVideoToVideoPipeline, but it seems that AnimateDiffVideoToVideoPipeline doesn't exist in diffusers lib V0.30.3

from diffusers import AnimateDiffVideoToVideoControlNetPipeline
ImportError: cannot import name 'AnimateDiffVideoToVideoControlNetPipeline' from 'diffusers' (/data/hli358/envs/animatediff/lib/python3.10/site-packages/diffusers/__init__.py)

**Comment by a-r-r-o-w** [comment]
> AnimateDiffVideoToVideoPipeline doesn't exist in diffusers lib V0.30.3

This pipeline was not shipped with 0.30.3. The only changes in that release were CogVideoX vid2vid and img2vid.

For now, you will have to install diffusers from the `main` branch to be able to use this pipeline. So, `pip install git+https://github.com/huggingface/diffusers` should fix this error hopefully.

**Comment by HanLiii** [comment]
@a-r-r-o-w, Thank you for the help! I wonder if the AnimateDiffVideoToVideoControlNetPipeline can support the tile controlnet from [,](https://huggingface.co/lllyasviel/control_v11f1e_sd15_tile), similar to how it supports the OpenPose ControlNet in the example:

controlnet = ControlNetModel.from_pretrained(
     "lllyasviel/sd-controlnet-openpose", torch_dtype=torch.float16
 )
 
 If it currently doesn't support this, is there is a way to modify the pipeline code to add support for it?

**Comment by a-r-r-o-w** [comment]
Our ControlNetModel supports all the available controlnets I think. cc @asomoza in case I'm wrong.

I believe it should work smoothly but if you're facing any errors with loading/inference, feel free to open a new issue and we can try and help there (so that others from the diffusers team can take a look too)

**Comment by HanLiii** [comment]
Thank you for the clarification! The tile ControlNet loads smoothly into the pipeline, but it seems that the tile ControlNet isn’t having the expected effect in the output. Below are the results for comparison:

	•	Tile ControlNet output (using script from lllyasviel/control_v11f1e_sd15_tile):
TileControlnetOutput


	•	AnimateDiffVideoToVideoControlNetPipeline with Tile ControlNet output:
AnimateDiffVideoToVideoControlNetPipeline_TileControlnet


For context, the second test uses the input video:
https://github.com/user-attachments/assets/34f0aa12-c0b6-4b62-a43d-d8aaa7676c70

The first test uses the first frame of the input video, and both tests use the same prompt: “best quality, astronaut in space, dancing.” The output from the AnimateDiffVideoToVideoControlNetPipeline appears noticeably more blurry when using the tile ControlNet. Could you provide any guidance on how to resolve this issue?

**Comment by asomoza** [comment]
yes, we support all available controlnets for SD 1.5. Personally I haven't tested the SD 1.5 Tile one, not even with a single image. 

To understand better, the comparison you're doing is between the single image and with animatediff right? I've never seen someone using the Tile for animations, maybe you can test it with something like ComfyUI to see if we have something wrong or it's just that the Tile + AnimateDiff combination doesn't work.

**Comment by a-r-r-o-w** [comment]
The AnimateDiff motion adapters are known to have sort of a blurring effect and poorer quality when it comes to following image/video condition. I can do some tests soon when I find time to help you more on this.

Just curious, do you notice this behaviour when using AnimateLCM with tile controlnet? If not, it might be because the original motion adapters are not the best when it comes to high resolution animation quality (even with a controlnet). Typically, you have to involve more tricks like latent upscaling, unsampling, adetailer, etc. for good results.

**Comment by reallyigor** [comment]
Hey @HanLiii ,
Based on the results you attached cfg scale is probably set too high -> the image looks oversaturated.
+ I would definitely try using the original noise scheduler, not an LCM

**Comment by HanLiii** [comment]
@reallyigor @a-r-r-o-w @asomoza, thanks for the reply!

In my earlier comment, I included outputs from the AnimateDiffVideoToVideoControlNetPipeline using the OpenPose ControlNet. The results were vivid and detailed, which was encouraging.

However, when I use the same pipeline with the Tile ControlNet, the outputs appear blurry and lack the vivid details present in the OpenPose results.

To troubleshoot, I tried:

	Adjusted guidance_scale from 7.5 down to 3.0.
	Replaced LCMScheduler with DDIMScheduler.
	Removed the motion module to see if it affected the output quality.

However, the outputs remain blurry. Here’s an example using Tile ControlNet with guidance_scale=3.0 and DDIMScheduler:

	•	AnimateDiffVideoToVideoControlNetPipeline with Tile ControlNet output, gudiance_scale=3.0, DDIMScheduler:
	
DDIMScheduler_guidance3 0

**Pull request merged**
